internal/character/clara/ult.go

```diff
@@ -0,0 +1,79 @@
+package clara
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	Ult        key.Modifier = "clara-ult" // MAvatar_Klara_00_Ultra_WarriorMode
+	UltCounter key.Modifier = "clara-ult-enhanced-counter"
+)
+
+type ultCounterNum int
+
+func init() {
+	modifier.Register(Ult, modifier.Config{
+		BehaviorFlags: []model.BehaviorFlag{model.BehaviorFlag_STAT_BURST},
+		Stacking:      modifier.Refresh,
+		Duration:      2,
+		Listeners: modifier.Listeners{
+			OnAdd: func(mod *modifier.Instance) {
+				state := mod.State().(State)
+
+				// extra res
+				mod.AddProperty(prop.AllDamageReduce, ultCut[state.ultLevelIndex])
+				// extra aggro
+				mod.AddProperty(prop.AggroPercent, 5)
+
+				amt := 2
+				if mod.Engine().HasModifier(mod.Owner(), E6) {
+					amt = 3
+				}
+
+				mod.Engine().AddModifier(mod.Owner(), info.Modifier{
+					Name:   UltCounter,
+					Source: mod.Owner(),
+					State:  ultCounterNum(amt),
+				})
+			},
```

Comment:
* You are not setting state in `AddModifier` call so this logic fails
* Nothing in this logic seems to necessitate a `State` + `OnAdd` listener. Should be able to just set `Stats` and put this logic in the `Ult` function directly (so 1. create `clara-ult-buff`, 2. create `enhanced-counter` buff)
* When doing this directly in ult, you do not need to do the expensive `E6` modifier check and can just check her `eidolon` level directly
* `clara-ult-buff` needs to be registered as `StatusType == BUFF`